src/System.Text.Json/tests/Serialization/Stream.ReadTests.cs

```diff
@@ -46,5 +63,21 @@ public static async Task ReadPrimitivesAsync()
                 Assert.Equal(1, i);
             }
         }
+
+        [Fact]
+        public static async Task ReadPrimitivesWithTrailingTriviaAsync()
+        {
+            using (MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes(" 1\t// Comment\r\n/* Multi\r\nLine */")))
+            {
+                JsonSerializerOptions options = new JsonSerializerOptions
+                {
+                    DefaultBufferSize = 1,
+                    ReadCommentHandling = JsonCommentHandling.Skip,
```

Comment:
I would add a test for "Allow comments" as well (especially for more than 1 leading/trailing comment).

Reply:
If you nest complex documents, JsonSerializer uses JsonDocument which throws:
```
System.Text.Json.Serialization.Tests.ObjectTests.ReadClassIgnoresLeadingOrTrailingTrivia(leadingTrivia: "/* Multi\nLine\nComment */ ", trailingTrivia: "\t// trailing comment\n ") [FAIL]
      System.ArgumentException : Comments cannot be stored in a JsonDocument, only the Skip and Disallow comment handling modes are supported.
      Parameter name: reader
```

Reply:
Good point. In that case, can we add a test for simple POCO (that doesn't have `System.Object` as a property), with allow comments mode?

@steveharter, @JeremyKuhne, @bartonjs  - this is something we should fix up. Either the serialize should disallow "allow comments" similar to `JsonDocument` OR convert it to skip before passing it to JsonDocument since the "allow" mode doesn't make much sense in this context.

@watfordgnf, this can be deferred for now (i.e. the PR doesn't need to block on this discussion).

Reply:
Sounds good.